From 66b27d3a5eae7d6b424b89414bcf952b971a93f6 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Wed, 16 Nov 2011 00:38:42 +0800
Subject: [PATCH 173/280] ENGR00162194 vpu: Initialize vpu registers to 0
 before vpu running

Initialize vpu registers to 0 before vpu running to avoid abnormal
case.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/vpu_lib.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index 1037f37..7f45087 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -177,6 +177,9 @@ RetCode vpu_Init(void *cb)
 		}
 
 		IOClkGateSet(true);
+
+		for (i = 0; i < 64; i++)
+			VpuWriteReg(BIT_CODE_BUF_ADDR + (i * 4), 0);
 		VpuWriteReg(BIT_PARA_BUF_ADDR, paraBuffer);
 		VpuWriteReg(BIT_CODE_BUF_ADDR, codeBuffer);
 		VpuWriteReg(BIT_TEMP_BUF_ADDR, tempBuffer);
-- 
1.8.0

